test/BingoGame.Indexer.CA.Tests/Processors/PlayedLogEventProcessorTests.cs

```diff
@@ -0,0 +1,104 @@
+        const string chainId = "AELF";
+        const string blockHash = "3c7c267341e9f097b0886c8a1661bef73d6bb4c30464ad73be714fdf22b09bdd";
+        const string previousBlockHash = "9a6ef475e4c4b6f15c37559033bcfdbed34ca666c67b2ae6be22751a3ae171de";
+        const string transactionId = "c09b8c142dd5e07acbc1028e5f59adca5b5be93a0680eb3609b773044a852c43";
+        const long blockHeight = 200;
+        var blockStateSetAdded = new BlockStateSet<LogEventInfo>
+        {
+            BlockHash = blockHash,
+            BlockHeight = blockHeight,
+            Confirmed = true,
+            PreviousBlockHash = previousBlockHash
+        };
+        
+        var blockStateSetTransaction = new BlockStateSet<TransactionInfo>
+        {
+            BlockHash = blockHash,
+            BlockHeight = blockHeight,
+            Confirmed = true,
+            PreviousBlockHash = previousBlockHash
+        };
+
+        var blockStateSetKey = await InitializeBlockStateSetAsync(blockStateSetAdded, chainId);
+        var blockStateSetKeyTransaction = await InitializeBlockStateSetAsync(blockStateSetTransaction, chainId);
+                //step2: create logEventInfo
+        var bingoed = new Played
+        {
+            PlayBlockHeight = blockHeight,
+            PlayerAddress = Address.FromPublicKey("AAA".HexToByteArray()),       
+            Amount = 100000000,
+            Type = BingoType.Large,
+            PlayId = HashHelper.ComputeFrom("PlayId"),
+            Symbol = "ELF",
+        };
+        var logEventInfo = LogEventHelper.ConvertAElfLogEventToLogEventInfo(bingoed.ToLogEvent());
+        logEventInfo.BlockHeight = blockHeight;
+        logEventInfo.ChainId = chainId;
+        logEventInfo.BlockHash = blockHash;
+        logEventInfo.TransactionId = transactionId;
+        var logEventContext = new LogEventContext
+        {
+            ChainId = chainId,
+            BlockHeight = blockHeight,
+            BlockHash = blockHash,
+            PreviousBlockHash = previousBlockHash,
+            TransactionId = transactionId,
+            Params = "{ \"to\": \"ca\", \"symbol\": \"ELF\", \"amount\": \"100000000000\" }",
+            To = "CAAddress",
+            MethodName = "Played",
+            ExtraProperties = new Dictionary<string, string>
+            {
+                { "TransactionFee", "{\"ELF\":\"30000000\"}" },
+                { "ResourceFee", "{\"ELF\":\"30000000\"}" }
+            },
+            BlockTime = DateTime.UtcNow
+        };
+        var bingoedLogEventProcessor = GetRequiredService<PlayedProcessor>();
+        
+        await bingoedLogEventProcessor.HandleEventAsync(logEventInfo, logEventContext);
+        
+        bingoedLogEventProcessor.GetContractAddress(chainId);
+        
+        //step4: save blockStateSet into es
+        await BlockStateSetSaveDataAsync<LogEventInfo>(blockStateSetKey);
+        await BlockStateSetSaveDataAsync<TransactionInfo>(blockStateSetKeyTransaction);
+        await Task.Delay(2000);
```

Comment:
Why do we need to do it here? What happens if we don't do it? How is it done in production?

Reply:
this delay is to make sure data being stored in local ES, then we can verify if data in ES is correct, e.g. bingoGameIndexData.Amount.ShouldBe(100000000); in live env, we don't need to verify data in ES, this verification only happens in test cases